extensions/narayana-jta/runtime/src/main/java/io/quarkus/narayana/jta/runtime/TransactionConfiguration.java

```diff
@@ -0,0 +1,27 @@
+package io.quarkus.narayana.jta.runtime;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Inherited;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * This annotation can be used to configure a different transaction timeout that the default one for a method or a class.
+ * It needs to be used on the entry method of the transaction.
```

Comment:
It can be defined on a class but needs to be defined on the entry method of the transaction. I think some information about how it works at class level is needed.

Reply:
I'm agree that it's not very clear and the transactional guide also is not very specific about this as it also states "entry method of a transaction".

Is it more clear if I add a note in both the guide and the JavaDoc saying
"If set on a class, it is equivalent that setting it on all methods of the class"

I can also replace "entry method" by "entry point" in the corresponding JavaDoc.

Reply:
``If defined on a class, it is equivalent to defining it on all the methods of the class marked with `@Transactional`.``
maybe?

I think I would go with `the method starting the transaction` instead of `the entry method`. That would be clearer.

Reply:
OK, do I also replace the sentence on the line 40 of the guide ?

Reply:
@loicmathieu could you point me to the line you're talking about?

Reply:
@gsmet Line 40 of the transactional guide:
> The easiest way to define your transaction boundaries is to use the `@Transactional` annotation on your entry method (`javax.transaction.Transactional`).

Reply:
OK, well, let's leave it at that then if it was a previously accepted wording.